sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SQLHadoopMapReduceCommitProtocol.scala

```diff
@@ -66,4 +91,37 @@ class SQLHadoopMapReduceCommitProtocol(
     logInfo(s"Using output committer class ${committer.getClass.getCanonicalName}")
     committer
   }
+
+  /**
+   * Called on the driver after a task commits. This can be used to access task commit messages
+   * before the job has finished. These same task commit messages will be passed to commitJob()
+   * if the entire job succeeds.
+   * Override it to check dynamic partition limitation on driver side.
+   */
+  override def onTaskCommit(taskCommit: TaskCommitMessage): Unit = {
```

Comment:
this implementation completely hides org.apache.spark.internal.io.HadoopMapReduceCommitProtocol#commitTask, which was the behaviour earlier.

Is it intensional?

Reply:
> this implementation completely hides org.apache.spark.internal.io.HadoopMapReduceCommitProtocol#commitTask

No. `onTaskCommit`  doesn't hide `commitTask`. Actually, `commitTask` is called on executor side but `onTaskCommit` is called on driver side.

Reply:
`SQLHadoopMapReduceCommitProtocol.onTaskCommit` overrides `FileCommitProtocol.onTaskCommit` on purpose.